internal/services/network/client/client.go

```diff
@@ -15,6 +16,13 @@ import (
 type Client struct {
 	*network_2023_09_01.Client
 
+	// This additional version has been added to facilitate migration of the remaining network
+	// resources to `hashicorp/go-azure-sdk` because upgrading directly to 2023-09-01 will have introduced
+	// breaking changes
+	// TODO 4.0 the following resources/data sources should be updated to use 2023-09-01 just prior to releasing 4.0:
+	// `azurerm_application_gateway` (resource and data source)
+	V20220701Client *network_2022_07_01.Client
```

Comment:
Given [how large the `network` package is](https://github.com/hashicorp/go-azure-sdk/tree/main/resource-manager/network/2022-07-01) both for compile-time and binary-size reasons we should only import a single version of the Meta Client (since most things should be using the default version).

Instead where we need to import multiple API Versions, particularly for large API Versions (such as Compute/Network etc) - can we update this to import only the specific SDK clients/paths that we need from the older API Version?